src/main/java/fi/hsl/transitdata/railsource/RailSpecific.java

```diff
@@ -14,4 +14,21 @@
                 .filter(GtfsRealtime.FeedEntity::hasTripUpdate)
                 .collect(Collectors.toList());
     }
+
+    static GtfsRealtime.TripUpdate fixInvalidTripUpdateDelayUsage(GtfsRealtime.TripUpdate tripUpdate) {
+        //If trip update has specified delay and stop time updates, timing information in stop time updates is ignored
+        //-> remove delay from trip update
+        if (tripUpdate.hasDelay() && tripUpdate.getStopTimeUpdateCount() != 0) {
+            GtfsRealtime.TripUpdate fixedTripUpdate = GtfsRealtime.TripUpdate.newBuilder()
```

Comment:
Maybe possible to do something like?:
`tripUpdate.toBuilder().setDelay(null).build()`
I don't know. Do you know?

Reply:
Seems to be possible to use `clearDelay()` to remove delay field. Fixed.